src/screens/UploadFiles.tsx

```diff
@@ -0,0 +1,193 @@
+import { ChangeEvent, useReducer, useState, useEffect } from "react";
+
+import ManifestDefinitions from "src/components/ManifestDefs";
+import { Button } from "../../src/components/Button";
+import { Alert } from "@us-gov-cdc/cdc-react";
+
+import * as tus from "tus-js-client";
+
+import { getEnv } from "../../src/utils/helperFunctions/env";
+
+interface FileUpload {
+  file: File;
+  manifest: string;
+}
+
+interface DispatchAction {
+  type: string;
+  value?: any;
+}
+
+function UploadFiles() {
+  const fileTypes = [".csv", ".hl7", ".txt"];
+
+  const initialState: FileUpload = {
+    file: new File([""], ""),
+    manifest: "",
+  };
+
+  const [uploadFeedback, setUploadFeedback] = useState<Array<string>>(["", ""]);
+
+  function reducer(state: FileUpload, action: DispatchAction) {
+    switch (action.type) {
+      case "updateFile":
+        return {
+          ...state,
+          file: action.value,
+        };
+      case "updateManifest":
+        return {
+          ...state,
+          manifest: action.value,
+        };
+      case "reset":
+        return initialState;
+      default:
+        throw new Error("Unrecognized action type provided to form reducer");
+    }
+  }
+
+  const [formState, dispatch] = useReducer(reducer, initialState);
+
+  const handleFileSelection = () => {
+    document?.getElementById("file-uploader")?.click();
+  };
+
+  const handleFileNameChange = (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.target.files) {
+      dispatch({
+        type: "updateFile",
+        value: e.target.files[0],
+      });
+    }
+  };
+
+  const handleManifestInputChange = (e: ChangeEvent<HTMLTextAreaElement>) => {
+    dispatch({
+      type: "updateManifest",
+      value: e.target.value,
+    });
+  };
+
+  const handleReset = () => {
+    dispatch({ type: "reset" });
+
+    // Manually reset file input since reselecting same file wont trigger onChange
+    const inputElement = document.getElementById(
+      "file-uploader"
+    ) as HTMLInputElement;
+    if (inputElement) inputElement.value = ""; // Set to an empty string to clear
+  };
+
+  // Todo: Disable submit when one is uploading
```

Comment:
How do you mean "one is uploading"? When a user is uploading or when one file is uploading?